contentcuration/contentcuration/frontend/channelEdit/components/ContentNodeListItem/index.vue

```diff
@@ -307,6 +332,41 @@
       },
     },
     methods: {
+      ...mapActions(['showSnackbar', 'clearSnackbar']),
+      ...mapActions('contentNode', ['deleteContentNode', 'copyContentNode']),
+      removeFailedCopyNode() {
+        return this.deleteContentNode(this.node.id);
+      },
+      retryFailedCopy: withChangeTracker(async function(changeTracker) {
+        const original_source_node = await ContentNode.table
+          .where('node_id')
+          .equals(this.node.original_source_node_id)
+          .toArray();
+        this.showSnackbar({
+          duration: null,
+          text: this.$tr('creatingCopies'),
+        });
+
+        return this.copyContentNode({
+          id: original_source_node[0]['id'],
+          target: this.node.id,
+          position: RELATIVE_TREE_POSITIONS.RIGHT,
+          wait_for_status: true,
+          startingRev: changeTracker._startingRev,
+          is_retry: true,
+        })
+          .then(() => {
+            this.showSnackbar({
+              text: this.$tr('copiedSnackbar'),
+              actionText: this.$tr('undo'),
+              actionCallback: () => changeTracker.revert(),
+            }).then(() => changeTracker.cleanUp());
+          })
+          .catch(() => {
+            this.clearSnackbar();
+            changeTracker.cleanUp();
+          });
+      }),
```

Comment:
Okay sir, will update.

I know there is a reason..? Ummm, what's the exact reason sir? :thinking:

Reply:
A channel can be view-only, so this component is used when that's the case. It's unlikely there's actually a scenario where a view-only user could see a failed copy, but since the `ContentNodeEditListItem.vue` component depends on this, I think it's fine to have it render the error, but not allow any actions.